Asterix/optics/propagation_functions.py

```diff
@@ -414,16 +427,16 @@ def prop_angular_spectrum(pup, lam, z, rad, prad, gamma=2):
     Nfourier = int(gamma * diam_pup_in_pix)
     cycles = diam_pup_in_pix
 
-    four = np.fft.fft2(crop_or_pad_image(pup, Nfourier), norm='ortho')
+    four = np.fft.fft2(crop_or_pad_image(pup, Nfourier), norm='ortho').astype(dtype_complex)
     u, v = np.meshgrid(np.arange(Nfourier) - Nfourier / 2, np.arange(Nfourier) - Nfourier / 2)
 
     rho2D = np.fft.fftshift(np.hypot(v, u)) * (cycles / diam_pup_in_m) / Nfourier
 
-    angular = np.exp(-1j * np.pi * z * lam * (rho2D**2))
-    return np.fft.ifft2(angular * four, norm='ortho')
+    angular = np.exp(-1j * np.pi * z * lam * (rho2D**2), dtype=dtype_complex)
+    return np.fft.ifft2(angular * four, norm='ortho').astype(dtype_complex)
```

Comment:
Does the numpy fft not retain the bit number of its inputs?

Reply:
apparently not : https://github.com/numpy/numpy/issues/17801
for this reason, the time improvement due to this modification is probably limited in 2DMs case as we require angular spectrum that require fft currently

Reply:
understood